reactive/containerd.py

```diff
@@ -289,6 +290,7 @@ def proxy_changed():
 
 @when('containerd.ready')
 @when('endpoint.containerd.joined')
+@when_not('endpoint.containerd.departed')
```

Comment:
Can you be `joined` and `departed` at the same time?  This shouldn't have any negative affect as-is, but i'm curious if you can have a remote unit coming in and leaving:

https://github.com/juju-solutions/charms.reactive/blob/master/charms/reactive/endpoints.py#L46

Reply:
For posterity, we had a chat about this. In general, you can be joined to multiple remote units. When one leaves, you'll be both departed (from the leaver) and joined (to the remaining units).

In this case, however, containerd is a subordinate -- there will only be 1 principal on the other side. Once the `-departed` hook runs, the only related unit gets removed from `all_joined_units` and `joined` will get toggled off here(ish):

https://github.com/juju-solutions/charms.reactive/blob/master/charms/reactive/endpoints.py#L227

Soooo, in this case, this `when_not` is probably not needed, but it doesn't hurt like my head does right now.